test/ATokenCheckpointed.t.sol

```diff
@@ -0,0 +1,1974 @@
+// SPDX-License-Identifier: Unlicensed
```

Comment:
Most of these tests were just copied as-is from the naive implementation's tests. Here is a temporary diff to make review easier: https://www.diffchecker.com/A87zLOy3

It's just [this](https://raw.githubusercontent.com/ScopeLift/flexible-voting/7efdd7b8deabd91a4950fc02ab6ef659bac32156/test/ATokenCheckpointed.t.sol) diff'ed against [this](https://raw.githubusercontent.com/ScopeLift/flexible-voting/1cd6967167928a48deb796c2c8f1a2d820a70d31/test/AaveAtokenFork.t.sol).